frontend/javascripts/oxalis/controller/combinations/bounding_box_handlers.ts

```diff
@@ -221,9 +225,57 @@ export function getClosestHoveredBoundingBox(
     secondaryEdge = getEdgeInfoFromId(secondNeighbourId);
   }
 
-  // @ts-expect-error ts-migrate(2322) FIXME: Type '{ boxId: number; direction: string; isMaxEdg... Remove this comment to see the full error message
   return [primaryEdge, secondaryEdge];
 }
+
+export function createBoundingBoxAndGetEdges(
+  pos: Point2,
+  plane: OrthoView,
+): [SelectedEdge, SelectedEdge | null | undefined] | null {
+  const state = Store.getState();
+  const globalPosition = calculateMaybeGlobalPos(state, pos, plane);
+
+  if (globalPosition == null) return null;
+
+  Store.dispatch(
+    addUserBoundingBoxAction({
+      boundingBox: {
+        min: globalPosition,
+        max: V3.add(globalPosition, [1, 1, 1], [0, 0, 0]),
+      },
+    }),
+  );
+  const { userBoundingBoxes } = getSomeTracing(Store.getState().tracing);
+
+  const indices = Dimension.getIndices(plane);
+  let newestBoundingBox =
+    userBoundingBoxes.length > 0 ? userBoundingBoxes[userBoundingBoxes.length - 1] : null;
+
+  if (newestBoundingBox == null) {
+    return null;
+  }
+
+  const nearestBoundingBox = newestBoundingBox;
+
+  const getEdgeInfoFromId = (edgeId: number) => {
+    const direction: SelectedEdge["direction"] = edgeId < 2 ? "horizontal" : "vertical";
+    const isMaxEdge = edgeId % 2 === 1;
+    const resizableDimension = direction === "horizontal" ? indices[1] : indices[0];
+    return {
+      boxId: nearestBoundingBox.id,
+      direction,
+      isMaxEdge,
+      edgeId,
+      resizableDimension,
+    };
+  };
```

Comment:
This method doesn't seem to differ from the earlier one and could be extracted if indices and nearestBoundingBox are supplied as arguments.